test/e2e/cluster_tls_endpoints.go

```diff
@@ -0,0 +1,204 @@
+				"clusterName":                 customerClusterName,
+				"managedResourceGroupName":    managedResourceGroupName,
+				"nsgName":                     customerNetworkSecurityGroupName,
+				"subnetName":                  customerVnetSubnetName,
+				"vnetName":                    customerVnetName,
+				"userAssignedIdentitiesValue": userAssignedIdentities,
+				"identityValue":               identity,
+				"keyVaultName":                keyVaultName,
+				"etcdEncryptionKeyName":       etcdEncryptionKeyName,
+			},
+			45*time.Minute,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("ensuring the API TLS certificate issued is not an OpenShift root CA")
+		clusterResp, err := tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient().Get(ctx, *resourceGroup.Name, customerClusterName, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(clusterResp.Properties).NotTo(BeNil())
+		Expect(clusterResp.Properties.API).NotTo(BeNil())
+		Expect(clusterResp.Properties.API.URL).NotTo(BeNil())
+
+		apiServerURL := clusterResp.Properties.API.URL
+		actualAPICert, err := tlsCertFromURL(ctx, *apiServerURL)
+		Expect(err).NotTo(HaveOccurred())
+
+		fmt.Fprintf(GinkgoWriter, "Issuer: %v\n", actualAPICert.Issuer)
+		Expect(actualAPICert.Issuer).NotTo(SatisfyAll(
+			HaveField("CommonName", "root-ca"),
+			HaveField("OrganizationalUnit", ContainElements("openshift")),
+		), "expected certificate not issued by an OpenShift root CA")
+
+		By("creating the node pool")
+		_, err = framework.CreateBicepTemplateAndWait(ctx,
+			tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+			*resourceGroup.Name,
+			"node-pool",
+			framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/modules/nodepool.json")),
+			map[string]interface{}{
+				"openshiftVersionId": openshiftNodeVersionId,
+				"clusterName":        customerClusterName,
+				"nodePoolName":       customerNodePoolName,
+				"replicas":           2,
+			},
+			45*time.Minute,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("ensuring the ingress TLS certificate issued by an OpenShift root CA")
+		hcpOpenShiftClustersClient := tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient()
+
+		By("waiting for the console URL to become available")
+		var consoleURL string
+		Eventually(func() bool {
+			resp, err := hcpOpenShiftClustersClient.Get(ctx, *resourceGroup.Name, customerClusterName, nil)
+			if err != nil || resp.Properties == nil || resp.Properties.Console == nil || resp.Properties.Console.URL == nil {
+				return false
+			}
+			consoleURL = *resp.Properties.Console.URL
+			fmt.Fprintln(GinkgoWriter, "Ingress URL found:", consoleURL)
+			return true
+		}).WithTimeout(15 * time.Minute).WithPolling(10 * time.Second).Should(BeTrue())
+
+		By("examining the server certificate returned by the default ingress when routing the console URL")
+		// Wait for the certificate to be loaded after concole starts
+		time.Sleep(2 * time.Minute)
```

Comment:
can we not put a hard sleep in here and rather wait for the ingress controllers to restart?